### ISSUE TITLE
Stop modifying item.nodeid.

### ISSUE THREAD

**Issue by ericsnowcurrently** [tonybaloney/pytest-azurepipelines#20]
[Per the pytest folks](https://github.com/pytest-dev/pytest/issues/5259), modifying an item's node ID is not allowed.  However, the azurepipelines plugin [is doing so](https://github.com/tonybaloney/pytest-azurepipelines/blob/ce7600b/pytest_azurepipelines.py#L35)!  The problem is that [other plugins](https://github.com/microsoft/vscode-python/blob/77a510c/pythonFiles/testing_tools/adapter/pytest.py#L73) may rely on item node IDs.  The node IDs may also be needed to re-run tests.

We talked about this a little bit at PyCon and it sounds like the azurepipelines plugin has definite needs for which there isn't a hook that helps.  Hence changing the node ID is a hack to work around the problem.  I recommend opening an issue at https://github.com/pytest-dev/pytest about getting hooks added to meet your needs.

**Comment by freakboy3742** [comment]
For those who may be having this problem: PR #22 has a workaround for this. If you're affected (as I was), you can modify your `azure-pipelines.yml` to reference the PR branch. Replace the line that says:
```
pip install pytest pytest-azurepipelines
```
with
```
pip install pytest 
pip install git+https://github.com/freakboy3742/pytest-azurepipelines.git@default-nodeid
```

**Comment by tonybaloney** [comment]
Merged and release in 0.7.0

Sorry it took me so long to get to this
